tests/dbshell/test_mongodb.py

```diff
@@ -0,0 +1,175 @@
+import os
+import signal
+import subprocess
+import sys
+from pathlib import Path
+from unittest import mock
+
+from django_mongodb.client import DatabaseClient
+
+from django.db import connection
+from django.test import SimpleTestCase
+
+
+class MongoDBDbshellCommandTestCase(SimpleTestCase):
+    def settings_to_cmd_args_env(self, settings_dict, parameters=None):
+        if parameters is None:
+            parameters = []
+        return DatabaseClient.settings_to_cmd_args_env(settings_dict, parameters)
+
+    def test_fails_with_keyerror_on_incomplete_config(self):
+        with self.assertRaises(KeyError):
+            self.settings_to_cmd_args_env({})
+
+    def test_basic_params_specified_in_settings(self):
+        for options_parameters in [(None, None, None), ("value1", "value2", True)]:
+            with self.subTest(keys=options_parameters):
+                authentication_database, authentication_mechanism, retry_writes = (
+                    options_parameters
+                )
+                if authentication_database is not None:
+                    expected_args = [
+                        "mongosh",
+                        "--username",
+                        "someuser",
+                        "--password",
+                        "somepassword",
+                        "--authenticationDatabase",
+                        "value1",
+                        "--authenticationMechanism",
+                        "value2",
+                        "--retryWrites",
+                        "true",
+                        "mongodb://somehost:444/somedbname",
```

Comment:
I guess you didn't update this for the latest version which no longer uses URIs.